R/testing.R

```diff
@@ -0,0 +1,167 @@
+#' Create a kelp test harness object.
+#'
+#' Exposes same interface as [kelp()] but with mock responses, can be used
+#' in place of a real Kelp object for testing when it might be quicker to
+#' not have SeaweedFS itself running. Just backed by a local temp directory.
+#'
+#' @export
+#' @importFrom R6 R6Class
+kelp_harness <- R6::R6Class(
+  "kelp_harness",
+  cloneable = FALSE,
+
+  public = list(
+    #' @field seaweed_url Base URL for seaweed master. Can be a dummy value.
+    seaweed_url = NULL,
+
+    #' @description
+    #' Create a mock kelp client object for testing which doesn't require
+    #' a Seaweed instance to be running.
+    #'
+    #' @param seaweed_url Root URL of Seaweed (can be anything for testing)
+    #'
+    #' @return A new `kelp_harness` object
+    initialize = function(seaweed_url) {
+      self$seaweed_url <- seaweed_url
+      private$dir <- tempfile()
+      dir.create(private$dir)
+    },
+
+    #' @description
+    #' Upload file
+    #'
+    #' @param path Path to file to be uploaded
+    #' @param collection Collection name, acts as a namespace for files.
+    #'
+    #' @return The uploaded file ID.
+    upload_file = function(path, collection = NULL) {
+      id <- sprintf("%s,%s", sample.int(9, 1), ids::random_id(bytes = 5))
```

Comment:
Could you document the idea behind the scheme here? Why `[0-9],[[:xdigit:]]{10}`?